express/properties/non_scalar/bandgaps.py

```diff
@@ -63,24 +63,27 @@ def compute_on_mesh(self, type_="indirect"):
             gap, k1, k2 = self._find_gap(N_sk[0], ev_sk[0], ec_sk[0], type=type_)
         else:
             gap, k1, k2 = self._find_gap(N_sk.ravel(), ev_sk.ravel(), ec_sk.ravel(), type=type_)
-            k1 = divmod(k1, nk)
-            k2 = divmod(k2, nk)
+            k1 = divmod(k1, nk)[1]
+            k2 = divmod(k2, nk)[1]
 
         result = self._serialize_band_gaps(gap, type_)
         if k1 is not None and k2 is not None:
             result.update({
-                'kpointValence': self._round(self.ibz_k_points[k1[1] if isinstance(k1, tuple) else k1]),
-                'kpointConduction': self._round(self.ibz_k_points[k2[1] if isinstance(k2, tuple) else k2])
+                "kpointValence": self._round(self.ibz_k_points[k1]),
+                "kpointConduction": self._round(self.ibz_k_points[k2]),
+                "eigenvalueValence": self._get_eigenvalues_from_band(e_sk=ev_sk, k=k1),
```

Comment:
Just to confirm: we will have these per spin as well, like for the kpoints, right?

Reply:
Hmm good point. For now, we only ever return one direct and one indirect gap, right (even in the unrestricted case)? I was a bit confused by this (see question below). Should we call `compute_on_mesh` once per spin also?

Reply:
Indeed, looks like we ignore the spin for these transitions. Maybe we should retain the spin as well as the eigenvalue also, for clarity...

Reply:
Found an [informative StackExchange post](https://mattermodeling.stackexchange.com/questions/4032/spin-considerations-for-the-bandgap-and-optical-excitations#:~:text=What%20are%20the%20consequences%20for%20a%20bandgap%20if%20the%20conduction,implies%20having%20two%20different%20bandgaps.) on this topic